### PR TITLE
consistent autoscaling config among control plane components

### PR DESCRIPTION
**Pull request by linsun**
Currently mixer is configuring auto scale differently than pilot, gateway and ingress.  This PR makes the configuration consistent.

### CODE REVIEW

File: install/kubernetes/helm/subcharts/ingress/templates/autoscale.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.autoscaleMin }}
+{{- if .Values.autoscaleEnabled $spec.autoscaleMin $spec.autoscaleMax }}
```
Comment (ymesika) [inline install/kubernetes/helm/subcharts/ingress/templates/autoscale.yaml]: `and`?

File: install/kubernetes/helm/subcharts/ingress/templates/deployment.yaml [hunk 2]
```diff
@@ -10,8 +10,12 @@ metadata:
     release: {{ .Release.Name }}
     istio: ingress
 spec:
-{{- if not .Values.autoscaleMin }}
-  replicas: {{ .Values.replicaCount }}
+{{- if not $spec.autoscaleEnabled }}
```
Comment (ymesika) [inline install/kubernetes/helm/subcharts/ingress/templates/deployment.yaml]: `$spec` or `.Values`?

File: install/kubernetes/helm/subcharts/pilot/templates/deployment.yaml [hunk 3]
```diff
@@ -13,8 +13,12 @@ metadata:
   annotations:
     checksum/config-volume: {{ template "istio.configmap.checksum" . }}
 spec:
-{{- if not .Values.autoscaleMin }}
-  replicas: {{ .Values.replicaCount }}
+{{- if not $spec.autoscaleEnabled }}
```
Comment (ymesika) [inline install/kubernetes/helm/subcharts/pilot/templates/deployment.yaml]: `$spec`?

File: install/kubernetes/helm/subcharts/ingress/templates/deployment.yaml (hunk 2 above)
Comment (linsun) [reply install/kubernetes/helm/subcharts/ingress/templates/deployment.yaml]: oops, c-n-p error :(

**Comment by linsun** [comment]
@ymesika good catches!  Think I addresses all your comment and able to test it locally with `helm template`. PTAL

**Comment by linsun** [comment]
@wenchenglu can we get this merge in release-1.1?

**Comment by wenchenglu** [comment]
/lgtm
/approve
@linsun please go ahead and merge it once the required tests passed

**Comment by linsun** [comment]
merging as the test error has nothing to do w/ the PR.

**Pull request merged**